pkg/cache/util_test.go

```diff
@@ -0,0 +1,52 @@
+// Copyright 2022 TiKV Project Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cache
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestWeightAllocator(t *testing.T) {
+	t.Parallel()
+	re := require.New(t)
+
+	wa := NewWeightAllocator(1, 1)
+	re.Equal(1., wa.Get(0))
+
+	wa = NewWeightAllocator(10, 10)
+	for i := 0; i < 10; i++ {
+		re.Equal(1./55.*float64(10-i), wa.Get(i))
+	}
+
+	wa = NewWeightAllocator(10, 1)
+	re.Equal(1./10., wa.Get(0))
+
+	wa = NewWeightAllocator(10, 3)
+	re.Equal(1./21.*3., wa.Get(0))
```

Comment:
You should verify the wa.get(0) is twice thant the wa.get(6), and the mod one is same with the second.